### PR TITLE
Templates Added for Python3

### PR DESCRIPTION
**Pull request by arpitn30**
## Description
Added the templates for Python 3 under the directory xbmc/interfaces/python3 which will be used by Groovy to convert XML files into Python modules. The data type typemaps haven't been updated yet.

## Motivation and Context
This is a part of Kodi being ported to Python3 

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

@Razzeee

### CODE REVIEW

**Comment by Rechi** [comment]
The new python3 directory has to be added at https://github.com/xbmc/xbmc/blob/da31dde/cmake/treedata/optional/common/python.txt

File: xbmc/interfaces/python3/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            ContextItemAddonInvoker.h
+            LanguageHook.h
+            preamble.h
+            PyContext.h
+            PythonInvoker.h
+            pythreadstate.h
+            swig.h
+            XBPython.h)
+
+core_add_library(python_interface)
```
Comment (Rechi) [inline xbmc/interfaces/python3/CMakeLists.txt]: use `python3_interface` or something, same name can't be used twice

Comment (Rechi) [inline xbmc/interfaces/python3/CMakeLists.txt]: use `python3_interface` or something, same name can't be used twice

**Comment by razzeee** [comment]
Please update the copyright headers :)
From `Copyright (C) 2005-2013 Team XBMC` to `Copyright (C) 2005-2017 Team Kodi`

I'm not a fan of them either, but it's a common thing to do in xbmc land, especially if you create new files.

**Comment by razzeee** [comment]
And `http://xbmc.org` becomes `http://www.kodi.tv`

File: xbmc/interfaces/python3/CMakeLists.txt (hunk 1 above)
Comment (arpitn30) [reply xbmc/interfaces/python3/CMakeLists.txt]: Where will this library be added?

**Pull request merged**